great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -296,7 +291,30 @@ def save_expectation_suite(
         )
         if include_rendered_content:
             expectation_suite.render()
-        self.expectations_store.set(key, expectation_suite, **kwargs)
+
+        response = self.expectations_store.set(key, expectation_suite, **kwargs)  # type: ignore[func-returns-value]
+        if isinstance(response, GeCloudResourceRef):
+            expectation_suite.ge_cloud_id = response.ge_cloud_id
```

Comment:
I was going to return it but since that violates the signature and behavior set throughout the inheritance hierarchy, I opted to update the object in-place. The tests ensure that we have coverage over the id updating the input object.

Reply:
Makes sense. Thanks for clarifying!